### ISSUE TITLE
[Extensibility-bug] Posted Invt. Document Print Action

### ISSUE THREAD

**Issue by vlkov** [microsoft/ALAppExtensions#15436]
Hi, 

When we print Posted Inventory Shipment or Posted Inventory Receipt the record number is not applied as a filter in the report.
We need to change it like in the unposted document.
The Print action on pages Posted Invt. Shipment, Posted Invt. Shipments,Posted Invt. Receipts,Posted Invt. Receipts,Posted Invt. Receipt should be like this, without another unassigned variable.

` trigger OnAction()`
  `              begin`
  `                  Rec.PrintRecords(true);`
 `               end;`

Thank you!

**Comment by Oleg-Romashkov** [comment]
Hi,

It is not clear what is the problem with typical approach of printing documents used all over the application. Could you provide a detailed scenario?

**Comment by vlkov** [comment]
Hi @Oleg-Romashkov,

Scenario:
1. Open Posted Inventory Shipment document
2. Click on action Print
3. Report dataitem is not filtered by selected document
![image](https://user-images.githubusercontent.com/43178739/154222905-f818cb65-9241-4ccc-892f-8aa7fcccf918.png)

It's happened because in codeunit 229 "Document-Print" functions PrintInvtShipment has parametr NewInvtShipmentHeader without VAR.

Posted Inventory Receipt has the same situation.

**Comment by Oleg-Romashkov** [comment]
What is the structure of your report? It should have a top level dataitem related to "Invt. Shipment Header".
For example, following report
```

report 50000 MyReport
{
    ProcessingOnly = true;

    dataset
    {
        dataitem(DataItemName; "Invt. Shipment Header")
        {
            column(ColumnName; "No.")
            {

            }
        }
    }
}

```
receives the "No." filter value:
![image](https://user-images.githubusercontent.com/82887855/154228297-cf9bed90-f06c-4b0b-99b8-1e8af49f1144.png)

**Comment by vlkov** [comment]
![image](https://user-images.githubusercontent.com/43178739/154229734-210b5294-efd9-4371-8935-29769976327e.png)

Yes, I'm sure that I have the correct report structure.
Let's go to code.
First look on page 6570 "Posted Invt. Shipment" Print action
`trigger OnAction()`
`                var`
`                    InvtShptHeader: Record "Invt. Shipment Header";`
`                begin`
`                    CurrPage.SetSelectionFilter(InvtShptHeader);`
`                    InvtShptHeader.PrintRecords(true);`
`                end;`
Here we see that we mark selected records at variable InvtShptHeader.
Next in fuction PrintRecords we send this variable to DocumentPrint.PrintInvtShipment without VAR 
In function DocumentPrint.PrintInvtShipment we have code
`  InvtShipmentHeader.Copy(NewInvtShipmentHeader);`
`        InvtShipmentHeader.SetRecFilter();`
NewInvtShipmentHeader is empty and when we call SetRecFilter()  record will filter by empty value

**Comment by Oleg-Romashkov** [comment]
Now the action code is:
```

                trigger OnAction()
                var
                    InvtShptHeader: Record "Invt. Shipment Header";
                begin
                    InvtShptHeader := Rec;
                    CurrPage.SetSelectionFilter(InvtShptHeader);
                    InvtShptHeader.PrintRecords(true);
                end;

```
This bug was fixed recently. Fixed code should be available on 19.4 which will be released soon.

**Comment by vlkov** [comment]
Fine, thank you!
